sdk/Trace/TraceContext.php

```diff
@@ -56,60 +54,44 @@ public static function extract($carrier, API\PropagationGetter $getter): API\Spa
     {
         $traceparent = $getter->get($carrier, self::TRACEPARENT);
         if ($traceparent === null) {
-            throw new \InvalidArgumentException('Traceparent not present');
+            return SpanContext::getInvalid();
         }
 
         // Traceparent = {version}-{trace-id}-{parent-id}-{trace-flags}
         $pieces = explode('-', $traceparent);
 
+        // Unable to extract traceparent. Expected 4 values
         $piecesCount = count($pieces);
-        if ($piecesCount != 4) {
-            throw new \InvalidArgumentException(
-                sprintf('Unable to extract traceparent. Expected 4 values, got %d', $piecesCount)
-            );
+        if ($piecesCount !== 4) {
+            return SpanContext::getInvalid();
         }
 
         $version = $pieces[0];
-        if ((preg_match(self::VALID_VERSION, $version) === 0) || ($version !== self::VERSION)) {
-            throw new \InvalidArgumentException(
-                sprintf('Only version 00 is supported, got %s', $version)
-            );
-        }
-
         $traceId = $pieces[1];
-        if ((preg_match(SpanContext::VALID_TRACE, $traceId) === 0) || ($traceId === SpanContext::INVALID_TRACE)) {
-            throw new \InvalidArgumentException(
-                sprintf('TraceID must be exactly 16 bytes (32 chars) and at least one non-zero byte, got %s', $traceId)
-            );
-        }
-
         $spanId = $pieces[2];
-        if ((preg_match(SpanContext::VALID_SPAN, $spanId) === 0) || ($spanId === SpanContext::INVALID_SPAN)) {
-            throw new \InvalidArgumentException(
-                sprintf('SpanID must be exactly 8 bytes (16 chars) and at least one non-zero byte, got %s', $spanId)
-            );
-        }
-
         $traceFlags = $pieces[3];
-        if (preg_match(self::VALID_TRACEFLAGS, $traceFlags) === 0) {
-            throw new \InvalidArgumentException(
-                sprintf('TraceFlags must be exactly 1 bytes (1 char) representing a bit field, got %s', $traceFlags)
-            );
-        }
-
-        // Only the sampled flag is extracted from the traceFlags (00000001)
-        $convertedTraceFlags = hexdec($traceFlags);
-        $isSampled = ($convertedTraceFlags & SpanContext::SAMPLED_FLAG) === SpanContext::SAMPLED_FLAG;
-
-        // Tracestate = 'Vendor1=Value1,...,VendorN=ValueN'
-        $rawTracestate = $getter->get($carrier, self::TRACESTATE);
-        if ($rawTracestate !== null) {
-            $tracestate = new TraceState($rawTracestate);
 
-            return SpanContext::restore($traceId, $spanId, $isSampled, true, $tracestate);
+        // Validates the version, traceId, spanId and traceFlags 
+        // If successful, returns a new span context with new context
+        if ($version === self::VERSION && SpanContext::isValidTraceId($traceId) && 
+            SpanContext::isValidSpanId($spanId) && SpanContext::isValidTraceFlag($traceFlags)) {
```

Comment:
I would invert this check.  Return fast if there is an invalid identifier and then the rest of the method is indented one level less, making it easier to follow.

Reply:
Fixed!